### ISSUE TITLE
[Bug] grad_norm=0 when gradient_accumulation_steps > 1

### ISSUE THREAD

**Issue by huyiwen** [fla-org/flame#20]
### Describe the Bug

Hi @yzhangcs, I found that when enabling gradient accumulation, the gradient norm is always zero.

![Image](https://github.com/user-attachments/assets/be534c25-fb79-435a-ad8b-c2636f50ebaf)

I found that the issue comes from `dist_utils.clip_grad_norm_`. It’s computing the norm over an empty list `grads`:

```python
@torch.no_grad()
def clip_grad_norm_(
    parameters: Union[torch.Tensor, Iterable[torch.Tensor]],
    max_norm: float,
    norm_type: float = 2.0,
    error_if_nonfinite: bool = False,
    foreach: Optional[bool] = None,
    pp_mesh: Optional[DeviceMesh] = None,
) -> torch.Tensor:

    grads = [p.grad for p in parameters if p.grad is not None]
    # len(grads) is 0
    ...
```

### Steps to Reproduce the Bug

set `gradient_accumulation_steps` greater than 1

only fsdp2 enabled with dp=16

### Expected Behavior

when `gradient_accumulation_steps == 1`:

![Image](https://github.com/user-attachments/assets/e15d944e-3af1-4298-85aa-be5209db7201)

### Environment Information

1. Torch: 2.6.0
2. Triton: 3.2.0

**Comment by yzhangcs** [comment]
@huyiwen Thank you!

**Comment by huyiwen** [comment]
It turns out to be related to https://github.com/pytorch/torchtitan/issues/990
